S4TF_Tutorial_6.ipynb

```diff
@@ -0,0 +1,1356 @@
+      },
+      "source": [
+        "# Data Preparation "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IQ3OILUgZeqb",
+        "colab_type": "text"
+      },
+      "source": [
+        "Images must be formatted into appropriately pre-processed floating point tensors before being fed into the network. The steps involved in preparing these images are:\n",
+        "\n",
+        "1. Read images from the disk\n",
+        "2. Decode contents of these images and convert it into proper grid format as per their RGB content\n",
+        "3. Convert them into floating point tensors\n",
+        "4. Rescale the tensors from values between 0 and 255 to values between 0 and 1, as neural networks prefer to deal with small input values.\n",
+        "5. Apply or not apply Image Augmentation Techniques, based on type of data\n",
+        "\n",
+        "We have done this in the following code."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0FyESvV0ZjhJ",
+        "colab_type": "text"
+      },
+      "source": [
+        "The `tensor` function takes 2 inputs are:\n",
+        "1. Image path as input\n",
+        "2. Whether or not to apply Image Augmentations\n",
+        "\n",
+        "The `tensors` function takes 3 inputs :\n",
+        "\n",
+        "1. List of paths\n",
+        "\n",
+        "2. Number of tensors to be produced in the output tensor as input\n",
+        "\n",
+        "3. Whether or not to apply Image Augmentations"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "3JihjHwPi2Ch",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "// Function `tensor(path:ImageAumentation:)` returns the resized image tensor w/ augmentation and it's corresponding label.\n",
+        "func tensor(path: String, ImageAumentation: Bool) -> (Tensor<Float>, Int32){\n",
+        "    var img = pil.Image.open(path)\n",
+        "    var image = np.array(img, dtype: np.float32) * (1.0 / 255)\n",
+        "    var imageTensor = Tensor<Float>(numpy: image)!\n",
+        "    \n",
+        "    if ImageAumentation == true {\n",
+        "        imageTensor = imageAugmentations(path: path)\n",
+        "    }\n",
+        "    \n",
+        "    imageTensor = imageTensor.expandingShape(at: 0)\n",
+        "    imageTensor = Raw.resizeArea(images: imageTensor , size: [150, 150])\n",
+        "    \n",
+        "    let label: Int32 = path.contains(\"dog.\") ? 0 : 1\n",
+        "    \n",
+        "    return (imageTensor, label)\n",
+        "}\n",
+        "\n",
+        "// Function `tensors(fromList:valueCount:ImageAumentation:)` returns the entire list converted to tensors of images and labels from `tensor(path:ImageAumentation:)` function.\n",
+        "func tensors(fromList: PythonObject, valueCount: Int, ImageAumentation: Bool) -> (Tensor<Float>, Tensor<Int32>) {\n",
```

Comment:
```suggestion
        "func images(fromList: PythonObject, valueCount: Int, augmented: Bool) -> (images: Tensor<Float>, labels: Tensor<Int32>) {\n",
```

Also, what does the "value" in `valueCount` stand for? Could you use a more descriptive name?

Reply:
Is `tensorCount` more appropriate? Or `imageCount`?

Reply:
`imageCount` is more descriptive here.